### PR TITLE
Add helper method for adding linking expectations in marker-based

### PR DESCRIPTION
**Pull request by GrahamPearsonAvaloq**
scoping tests.

### CODE REVIEW

File: com.avaloq.tools.ddk.xtext.test.core/src/com/avaloq/tools/ddk/xtext/test/scoping/AbstractScopingTest.java [hunk 1]
```diff
@@ -136,6 +168,17 @@ protected void beforeAllTests() {
     getTestInformation().putTestObject(Iterable.class, allContents);
   }
 
+  /**
+   * Test linking.
+   */
+  @Override
+  protected void afterEachTest() {
+    expectedLinks.forEach(ExpectedLink::test);
```
Comment (RGHenderson) [inline com.avaloq.tools.ddk.xtext.test.core/src/com/avaloq/tools/ddk/xtext/test/scoping/AbstractScopingTest.java]: I'd rather we test the links in some other method.

Comment (GrahamPearsonAvaloq) [reply com.avaloq.tools.ddk.xtext.test.core/src/com/avaloq/tools/ddk/xtext/test/scoping/AbstractScopingTest.java]: Done.

File: com.avaloq.tools.ddk.xtext.test.core/src/com/avaloq/tools/ddk/xtext/test/scoping/AbstractScopingTest.java [hunk 2]
```diff
@@ -641,6 +680,50 @@ public static String selectorFragmentSegment(final int containmentFeature, final
...
+   * target tag. Expectations can be tested by calling {@link #testExpectedLinking()}. Implicit items will be traversed.
+   *
+   * @see #mark(int)
+   * @see #testExpectedLinking()
+   * @param targetTag
+   *          Tag pointing to the destination object
+   * @return Mark text to be inserted in the source file, never {@code null}
+   */
+  protected String link(final int targetTag) {
+    return link(tag -> getObjectForTag(tag), targetTag);
```
[the remark this answers is not in the document]
Comment (GrahamPearsonAvaloq) [reply com.avaloq.tools.ddk.xtext.test.core/src/com/avaloq/tools/ddk/xtext/test/scoping/AbstractScopingTest.java]: Better still: I now store a collection of Runnables which each test an expected link.

File: com.avaloq.tools.ddk.xtext.test.core/src/com/avaloq/tools/ddk/xtext/test/scoping/AbstractScopingTest.java [hunk 3]
```diff
@@ -641,6 +680,50 @@ public static String selectorFragmentSegment(final int containmentFeature, final
...
+  }
+
+  /**
+   * Performs linking test. Checks expectations which were set using {@link #link(int)} or {@link #link(Function, int).
+   *
+   * @see #link(int)
+   * @see #link(Function<Integer, EObject>, int)
+   * @see #testLinking(int, EObject)
+   */
+  protected void testExpectedLinks() {
```
[the remark this answers is not in the document]
Comment (GrahamPearsonAvaloq) [reply com.avaloq.tools.ddk.xtext.test.core/src/com/avaloq/tools/ddk/xtext/test/scoping/AbstractScopingTest.java]: OK; done.

**Pull request merged**